pkg/vpc/vpc.go

```diff
@@ -524,16 +522,28 @@ func cleanupSubnets(spec *api.ClusterConfig) {
 	cleanup(&spec.VPC.Subnets.Public)
 }
 
-func validatePublicSubnet(subnets []*ec2.Subnet) error {
-	legacySubnets := make([]string, 0)
+func validatePublicSubnet(cfg *api.ClusterConfig, subnets []*ec2.Subnet) error {
+	containsSubnet := func(id string) bool {
+		for _, f := range cfg.VPC.Subnets.Public.WithIDs() {
+			if f == id {
+				return true
+			}
+		}
+		return false
+	}
+	var legacySubnets []string
 	for _, sn := range subnets {
 		if sn.MapPublicIpOnLaunch == nil || !*sn.MapPublicIpOnLaunch {
 			legacySubnets = append(legacySubnets, *sn.SubnetId)
 		}
+		if !containsSubnet(*sn.SubnetId) {
+			return fmt.Errorf("subnet %q does not exist or is not a public subnet", *sn.SubnetId)
+		}
```

Comment:
MEGA nit you could make map a `map[string]interface{}` once containing all the keys, so then each `contains` is just a map lookup instead of another loop over. but this doesn't matter at all :joy:

Reply:
I ACTUALLY THOUGHT ABOUT THAT! :DDDD 

But it's three or so subnets most of the time, so... :D It's not worth the extra lines + memory to store them in a map. :)